File: cordova-idcheckio/android/ParameterUtils.java
package com.ariadnext.idcheckio;

import com.ariadnext.idcheckio.sdk.bean.ConfirmationType;
import com.ariadnext.idcheckio.sdk.bean.DataRequirement;
import com.ariadnext.idcheckio.sdk.bean.DocumentType;
import com.ariadnext.idcheckio.sdk.bean.EnumExtraParameters;
import com.ariadnext.idcheckio.sdk.bean.Extraction;
import com.ariadnext.idcheckio.sdk.bean.FaceDetection;
import com.ariadnext.idcheckio.sdk.bean.FeedbackLevel;
import com.ariadnext.idcheckio.sdk.bean.FileSize;
import com.ariadnext.idcheckio.sdk.bean.Forceable;
import com.ariadnext.idcheckio.sdk.bean.Language;
import com.ariadnext.idcheckio.sdk.bean.Orientation;
import com.ariadnext.idcheckio.sdk.component.IdcheckioView;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Iterator;

public class ParameterUtils {

    public static void parseParameters(IdcheckioView.Builder idcheckioView, String params) throws JSONException{
        JSONObject jsonParams = new JSONObject(params);
        Iterator<String> keys = jsonParams.keys();
        while (keys.hasNext()) {
            String key = keys.next();
            String value = jsonParams.getString(key);
            switch (key) {
                case "DocumentType":
                    idcheckioView.docType(DocumentType.valueOf(value));
                    break;
                case "Orientation":
                    idcheckioView.orientation(Orientation.valueOf(value));
                    break;
                case "ConfirmType":
                    idcheckioView.confirmType(ConfirmationType.valueOf(value));
                    break;
                case "UseHd":
                    idcheckioView.useHd(Boolean.parseBoolean(value));
                    break;
                case "ScanBothSides":
                    idcheckioView.scanBothSides(Forceable.valueOf(value));
                    break;
                case "Side1Extraction":
                    JSONObject side1 = new JSONObject(value);
                    idcheckioView.sideOneExtraction(new Extraction(DataRequirement.valueOf(side1.getString("DataRequirement")),
                            FaceDetection.valueOf(side1.getString("FaceDetection"))));
                    break;
                case "Side2Extraction":
                    JSONObject side2 = new JSONObject(value);
                    idcheckioView.sideTwoExtraction(new Extraction(DataRequirement.valueOf(side2.getString("DataRequirement")),
                            FaceDetection.valueOf(side2.getString("FaceDetection"))));
                    break;
                case "ExtraParams":
                    JSONObject extraParams = new JSONObject(value);
                    Iterator<String> extraKeys = extraParams.keys();
                    while (extraKeys.hasNext()) {
                        String extraKey = extraKeys.next();
                        switch (extraKey) {
                            case "AdjustCrop":
                                idcheckioView.adjustCrop(Boolean.parseBoolean(extraParams.getString(extraKey)));
                                break;
                            case "Language":
                                idcheckioView.language(Language.valueOf(extraParams.getString(extraKey)));
                                break;
                            case "ManualButtonTimer":
                                idcheckioView.manualButtonTimer(Integer.parseInt(extraParams.getString(extraKey)));
                                break;
                            case "MaxPictureFilesize":
                                idcheckioView.maxPictureFilesize(FileSize.valueOf(extraParams.getString(extraKey)));
                                break;
                            case "FeedbackLevel":
                                idcheckioView.feedbackLevel(FeedbackLevel.valueOf(extraParams.getString(extraKey)));
                                break;
                            case "Token":
                                idcheckioView.token(extraParams.getString(extraKey));
                                break;
                            case "ConfirmAbort":
                                idcheckioView.confirmAbort(Boolean.parseBoolean(extraParams.getString(extraKey)));
                                break;
                            case "SdkEnvironment":
                                idcheckioView.extraParameter(EnumExtraParameters.SDK_ENVIRONMENT, extraParams.getString(extraKey));
                                break;
                            default:
                                break;
                        }
                    }
                    break;
                default:
                    break;
            }
        }
    }
}

File: cordova-idcheckio/android/ParameterUtils.kt
package com.ariadnext.idcheckio

import com.ariadnext.idcheckio.sdk.bean.*
import com.ariadnext.idcheckio.sdk.component.IdcheckioView
import com.ariadnext.idcheckio.sdk.interfaces.cis.CISType
import org.json.JSONException
import org.json.JSONObject

object ParameterUtils {
    @Throws(JSONException::class)
    fun parseParameters(idcheckioView: IdcheckioView.Builder, params: String) {
        val jsonParams = JSONObject(params)
        val keys = jsonParams.keys()
        while (keys.hasNext()) {
            val key = keys.next()
            val value = jsonParams.getString(key)
            when (key) {
                "docType" -> idcheckioView.docType(DocumentType.valueOf(value))
                "orientation" -> idcheckioView.orientation(Orientation.valueOf(value))
                "confirmationType" -> idcheckioView.confirmType(ConfirmationType.valueOf(value))
                "useHd" -> idcheckioView.useHd(value.toBoolean())
                "integrityCheck" -> {
                    val integrityCheck = JSONObject(value)
                    idcheckioView.integrityCheck(IntegrityCheck(readEmrtd = integrityCheck.getString("readEmrtd").toBoolean()))
                }
                "scanBothSides" -> idcheckioView.scanBothSides(Forceable.valueOf(value))
                "sideOneExtraction" -> {
                    val side1 = JSONObject(value)
                    idcheckioView.sideOneExtraction(Extraction(
                        DataRequirement.valueOf(side1.getString("codeline")), FaceDetection.valueOf(side1.getString("faceDetection")))
                    )
                }
                "sideTwoExtraction" -> {
                    val side2 = JSONObject(value)
                    idcheckioView.sideTwoExtraction(Extraction(
                        DataRequirement.valueOf(side2.getString("codeline")), FaceDetection.valueOf(side2.getString("faceDetection")))
                    )
                }
                "language" -> idcheckioView.language(Language.valueOf(value))
                "manualButtonTimer" -> idcheckioView.manualButtonTimer(value.toInt())
                "feedbackLevel" -> idcheckioView.feedbackLevel(FeedbackLevel.valueOf(value))
                "adjustCrop" -> idcheckioView.adjustCrop(value.toBoolean())
                "maxPictureFilesize" -> idcheckioView.maxPictureFilesize(FileSize.valueOf(value))
                "token" -> idcheckioView.token(value)
                "confirmAbort" -> idcheckioView.confirmAbort(value.toBoolean())
                "onlineConfig" -> {
                    val onlineJson = JSONObject(value)
                    val onlineConfig = OnlineConfig()
                    if(!onlineJson.isNull("isReferenceDocument")){
                        onlineConfig.isReferenceDocument = onlineJson.getString("isReferenceDocument").toBoolean()
                    }
                    if(!onlineJson.isNull("checkType")){
                        onlineConfig.checkType = CheckType.valueOf(onlineJson.getString("checkType"))
                    }
                    if(!onlineJson.isNull("cisType")){
                        onlineConfig.cisType = CISType.valueOf(onlineJson.getString("cisType"))
                    }
                    if(!onlineJson.isNull("folderUid")){
                        onlineConfig.folderUid = onlineJson.getString("folderUid")
                    }
                    if(!onlineJson.isNull("biometricConsent")){
                        onlineConfig.biometricConsent = onlineJson.getString("biometricConsent").toBoolean()
                    }
                    if(!onlineJson.isNull("enableManualAnalysis")){
                        onlineConfig.enableManualAnalysis = onlineJson.getString("enableManualAnalysis").toBoolean()
                    }
                    idcheckioView.onlineConfig(onlineConfig)
                }
                else -> { }
            }
        }
    }
}